src/foam/u2/search/GroupBySearchView.js

```diff
@@ -149,7 +149,7 @@ foam.CLASS({
     },
 
     function updatePredicate_(choice) {
-      var exists = typeof choice !== 'undefined' && choice !== '';
+      var exists = typeof choice !== 'undefined' && choice !== '' && choice != null;
       this.predicate = exists ? this.op.create({
```

Comment:
so there are at least two methods `findChoiceByData` and `findChoiceByText` in `foam2/src/foam/u2/view/ChoiceView.js` which set choice to null so added a null check

Reply:
as an option we probably could change those methods to return undefined

Reply:
I think that's a better idea, to modify those methods to return undefined.
commit f05febbd89e8defeaa67cd679f5b7ec830e8f5ff
Author: Kevin Greer <kgrgreer@gmail.com>
Date:   Fri Dec 18 10:34:45 2020 -0500

    ChoiceView findChoiceBy... methods now return undefined if not found instead of null.

Also, we should use foam.Undefined.isInstance() instead of typeof.